src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java

```diff
@@ -1898,6 +1774,146 @@ public XMPPError onRemoveSource(JingleSession sourceJingleSession,
             sourceJingleSession, sourcesToRemove, sourceGroupsToRemove, true);
     }
 
+    /**
+     * Updates the RTP description, transport and propagates sources and source
+     * groups of a participant that sends the session-accept or transport-accept
+     * Jingle IQs.
+     *
+     * @param jingleSession
+     * @param contents
+     * @return
+     */
+    private XMPPError updateParticipant(
+        JingleSession jingleSession,
+        List<ContentPacketExtension> contents)
+    {
+        Participant participant
+            = findParticipantForJingleSession(jingleSession);
+        Jid participantJid = jingleSession.getAddress();
+
+        if (participant == null)
+        {
+            String errorMsg
+                = "No participant found for: " + participantJid;
+            logger.warn(errorMsg);
+            return XMPPError.from(XMPPError.Condition.item_not_found,
+                errorMsg).build();
+        }
+
+        // XXX We will be acting on the received session-accept bellow.
+        // Unfortunately, we may have not received an acknowledgment of our
+        // session-initiate yet and whatever we do bellow will be torn down when
+        // the acknowledgement timeout occurs later on. Since we will have
+        // acted on the session-accept by the time the acknowledgement timeout
+        // occurs, we may as well ignore the timeout.
+        jingleSession.setAccepted(true);
```

Comment:
I don't understand much about jicofo state but is this really correct to do for transport-accept?

Reply:
that accepted flag is cleared on ice restart, and transport-accept is equivalent of 'session-accept' even though it may make no sense  from Jingle spec perspective. With 'transport-accept' the client will include the latest source description, so that's why the participant needs  to be updated. It wasn't a problem without Octo(I think), because those are fresh channels and jicofo makes sure that the previous ones have been expired by doing sync expire(due to out of order xmpp processing on the jvb).